File: py-101/py_101_109_small_problems/easy_2/when_will_i_retire.py
# Build a program that displays when the user will retire and how many years she has to work till retirement.

from datetime import datetime

def validate_input(message):
    while True:
        try:
            num_input = int(input(message))
            break
        except ValueError:
            print("Enter a valid integer.")

    return num_input

age = validate_input("What is your age? ")
retirement_age = validate_input("At what age would you like to retire? ")
years_to_go = max(0, retirement_age - age)

current_year = datetime.now().year

print(f"It's {current_year}. You will retire in {current_year + years_to_go}.")
print(f"You only have {years_to_go} years of work to go!")

File: js-210/exercises/small_problems/easy_3/when_will_i_retire.js
/*
Build a program that logs when the user will retire and how many more years the user has to work until retirement.

Example:

What is your age? 30
At what age would you like to retire? 70

It's 2017. You will retire in 2057.
You have only 40 years of work to go!
*/

const rlSync = require('readline-sync');

let age = parseInt(rlSync.question("What is your age? "));
let retirementAge = parseInt(rlSync.question("At what age would you like to retire? "));
let yearsOfWork = retirementAge - age;

let currentYear = new Date().getFullYear();
let retirementYear = currentYear + yearsOfWork;

console.log(`It's ${currentYear}. You will retire in ${retirementYear}.`);
console.log(`You have only ${yearsOfWork} years of work to go!`);